cl/opinion_page/forms.py

```diff
@@ -78,3 +95,255 @@ class DocketEntryFilterForm(forms.Form):
+            volume,
+            reporter,
+            page,
+        )
+
+    def verify_unique_judges(self):
+        if self.pk == "tennworkcompapp":
+            judges = [
+                self.cleaned_data["lead_author"],
+                self.cleaned_data["second_judge"],
+                self.cleaned_data["third_judge"],
+            ]
+            flag = len(set(judges)) == len(judges)
+            if not flag:
+                self.add_error(
+                    "lead_author",
+                    ValidationError("Please select each judge only once."),
+                )
+
+    def clean_pdf_upload(self):
+        pdf_data = self.cleaned_data.get("pdf_upload").read()
+        sha_1 = sha1(force_bytes(pdf_data))
+        exists = Opinion.objects.filter(sha1=sha_1).exists()
+        if exists:
+            op = Opinion.objects.get(sha1=sha_1)
+            self.add_error(
+                "pdf_upload",
+                ValidationError(
+                    format_html(
+                        'Document already in database. See: <a href="%s">%s</a>'
+                        % (op.get_absolute_url(), op.cluster.case_name),
+                    )
+                ),
+            )
+        return pdf_data
+
+    def make_panel(self):
+        if self.pk == "tennworkcompapp":
+            self.cleaned_data["panel"] = [
+                self.cleaned_data["lead_author"],
+                self.cleaned_data.get("second_judge"),
+                self.cleaned_data.get("third_judge"),
+            ]
+        else:
+            self.cleaned_data["panel"] = [self.cleaned_data["lead_author"]]
+
+    def make_item_dict(self):
+        self.cleaned_data["item"] = {
+            "source": Docket.DIRECT_INPUT,
+            "cluster_source": "D",
+            "case_names": self.cleaned_data.get("case_title"),
+            "case_dates": self.cleaned_data["publication_date"],
+            "precedential_statuses": "Published",
+            "docket_numbers": self.cleaned_data["docket_number"],
+            "judges": ", ".join(
+                [j.name_full for j in self.cleaned_data.get("panel")]
+            ),
+            "author_id": self.cleaned_data["lead_author"].id,
+            "author": self.cleaned_data["lead_author"],
+            "date_filed_is_approximate": "False",
+            "blocked_statuses": "False",
+            "neutral_citations": self.cleaned_data["neutral_citation"],
+            "download_urls": "",
+        }
+
+    def clean(self):
+        super(TennWorkersForm, self).clean()
+        self.validate_neutral_citation()
+        self.make_panel()
+        self.make_item_dict()
+        self.verify_unique_judges()
+        return self.cleaned_data
+
+    def save(self):
+        """Save uploaded Tennessee Workers Comp/Appeal to db.
+
+        :return: Cluster
+        """
+
+        sha1_hash = sha1(force_bytes(self.cleaned_data.get("pdf_upload")))
+        court = Court.objects.get(pk=self.cleaned_data.get("court_str"))
```

Comment:
I'm also annoyed that we have a `court` field we populated back in the `__init__` method. Don't we still have that available, avoiding us having to look up the Court object again?

Reply:
yes - its annoying but no we dont have it anymore.

Reply:
Atleast when I tested it.

Reply:
Doing this in the init - was done solely to simplify the publish.html code.

Reply:
OK.